dataset/src/main/scala/frameless/ops/ColumnTypes.scala

```diff
@@ -3,17 +3,25 @@ package ops
 
 import shapeless._
 
-trait ColumnTypes[V, U <: HList] {
+/** A type class to extract the column types out of an HList of [[frameless.TypedColumn]].
+  *
+  * @example
+  * {{{
+  *   type U = TypedColumn[T,A] :: TypedColumn[T,B] :: TypedColumn[T,C] :: HNil
+  *   type Out = A :: B :: C :: HNil
+  * }}}
+  */
+trait ColumnTypes[T, U <: HList] {
```

Comment:
Oh, just realized that this is a special case of using `Comapped`! Ok, I think I will add this comment in case someone in the future decide to revert this back to Comapped.